component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/client/EventHubClient.java

```diff
@@ -270,34 +270,48 @@ private EventDataBatch createBatchWithRetry() {
-                    producer.send(batch);
-                    batch = createBatchWithRetry();
-                    log.debug("Flushed " + size + " events to Analytics cluster.");
+                if (publishingLock.tryLock()) {
+                    try {
+                        int size = batch.getCount();
+                        producer.send(batch);
+                        batch = createBatchWithRetry();
+                        log.debug("[{ " + Thread.currentThread().getName().replaceAll("[\r\n]", "") + " }] "
+                                          + "Flushed " + size + " events to Analytics cluster.");
+                    } catch (Exception e) {
+                        log.error("Event flushing operation failed. Will be retried again according to the configured "
+                                          + "client.flushing.delay. Error will be handled by publishing threads once "
+                                          + "Event Data Batch is filled.");
+                        log.debug("[{ " + Thread.currentThread().getName().replaceAll("[\r\n]", "") + " }] "
+                                          + "Event flushing operation failed. Will be retried again according to the "
+                                          + "configured client.flushing.delay. Error will be handled by publishing "
+                                          + "threads once Event Data Batch is filled.", e);
+                        //Dont do anything for any exception. If it is recoverable exception next run will succeed.
+                        //If not recoverable then next run will be filtered by if condition
+                    } finally {
+                        publishingLock.unlock();
+                    }
                 } else {
-                    log.error("Event flushing operation aborted as publisher threads are trying to send events");
+                    log.debug("[{ " + Thread.currentThread().getName().replaceAll("[\r\n]", "") + " }] Event "
+                               + "flushing operation aborted as publisher threads are trying to send events");
                 }
-            } catch (Exception e) {
-                log.error("Event flushing operation failed. Will be retried again according to the configured "
-                                  + "client.flushing.delay. Error will be handled by publishing threads once "
-                                  + "Event Data Batch is filled.");
-                //Dont do anything for any exception. If it is recoverable exception next run will succeed.
-                //If not recoverable then next run will be filtered by if condition
-            } finally {
-                readWriteLock.writeLock().unlock();
-            }
         } else {
-            log.debug("Event flushing is aborted as Event Data Batch is empty or connection to Event Hub is not made");
+            log.debug("[{ " + Thread.currentThread().getName().replaceAll("[\r\n]", "") + " }] Event flushing "
```

Comment:
When we have non-trivial String operations, better to wrap the entire block with if(log.isDebugEnabled())

Reply:
fixed with 10529cd